content/en/docs/How-to guides/kaniko-build-push.md

```diff
@@ -0,0 +1,411 @@
+spec:
+  pipelineRef:
+    name: clone-read
+  workspaces:
+  - name: shared-data
+    volumeClaimTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+  params:
+  - name: repo-url
+    value: https://github.com/google/docsy-example.git
+```
+
+For this how-to we are using a public repository as an example. You can also
+[use *git clone* with private repositories, using SSH
+authentication][tekton-git-auth].
+
+[tekton-clone]: /docs/how-to-guides/clone-repository/ 
+[tekton-git-auth]: /docs/how-to-guides/clone-repository/#git-authentication
+[git-clone]: https://hub.tekton.dev/tekton/task/git-clone
+
+## Build the container image with Kaniko
+
+To build the image use the [Kaniko][kaniko] Task from the [community
+hub][tekton-hub]. 
+
+1.  Add the image reference to the `params` section in `pipeline.yaml`:
+
+    ```yaml
+    params: 
+    - name: image-reference
+      type: string
+    ```
+
+    This parameter is used to add the tag corresponding the container registry
+    where you are going to push the image. 
+
+1.  Create the new `build-push` Task in the same `pipeline.yaml` file:
+
+    ```yaml
+    tasks:
+    ...
+      - name: build-push
+        runAfter: ["fetch-repo"]
+        taskRef:
+          name: kaniko
+        workspaces:
+        - name: source
+          workspace: shared-data
+        params:
+        - name: IMAGE
+          value: $(params.image-reference)
+    ```
+
+    This new Task refers to `kaniko`, which is going to be installed from the
+    [community hub][tekton-hub]. A Task has its own set of `workspaces` and
+    `params` passed down from the parameters and Workspaces defined at Pipeline
+    level. In this case, the Workspace `source` and the value of `IMAGE`. Check
+    the [kaniko Task documentation][kaniko] to see all the available options.
+
+1.  Instantiate the `build-push` Task. Add the value of `image-reference` to
+    the `params` section in `pipelinerun.yaml`:
+
+    ```yaml
+    params:
+    - name: image-reference
+      value: container.registry.com/sublocation/my_app:version
+    ```
+
+    Replace `container.registry.com/sublocation/my_app:version` with the actual
+    tag for your registry. You can [set up a local registry][local-registry] for
+    testing purposes.
+
+Check the [full code samples](#full-code-samples) to see how all the pieces fit
+together.
+
+[kaniko]: https://hub.tekton.dev/tekton/task/kaniko
+[tekton-hub]: https://hub.tekton.dev/
+[local-registry]: https://gist.github.com/trisberg/37c97b6cc53def9a3e38be6143786589 
+
+### Container registry authentication
+
+In most cases, to push the image to a container registry you must provide
+authentication credentials first.
+
+1.  Set up authentication with the Docker credential helper and generate the
+    Docker configuration file, `~/.docker/config.json`, for your registry.
+    Check your cloud provider documentation to complete this step.
```

Comment:
Hmm, this is the only part of the tutorial that I find a bit confusing. As a nice to have, could we provide some more guidance here for some common registries e.g. docker hub, gcr, ghcr?

Reply:
For GCR at least I know that there is a credential helper tool that can make this easier when running in GKE: https://cloud.google.com/artifact-registry/docs/docker/authentication

Reply:
I wasn't sure about  that because of there being so many options and it will always feel like I leaving something out. I agree with you that it's helpful to have at least a few of them. I added a bunch of links for different registries.